tests/datasources/test_chirps.py

```diff
@@ -0,0 +1,329 @@
+"""Tests for the FewsNet module."""
+import calendar
+
+import cftime
+import numpy as np
+import pytest
+import xarray as xr
+from xarray.coding.cftimeindex import CFTimeIndex
+
+from aatoolbox import ChirpsDaily, ChirpsMonthly, GeoBoundingBox
+
+DATASOURCE_BASE_DIR = "chirps"
+_START_YEAR = 2020
+_START_MONTH = 7
+_START_DAY = 15
+_END_YEAR = 2020
+_END_MONTH = 8
+_END_DAY = 1
+_END_YEAR_FUTURE = 2100
+_END_MONTH_FUTURE = 10
+_END_DAY_FUTURE = 2
+
+_START_MONTH_STR = calendar.month_abbr[_START_MONTH]
+_END_MONTH_STR = calendar.month_abbr[_END_MONTH]
+_END_MONTH_FUTURE_STR = calendar.month_abbr[_END_MONTH_FUTURE]
+
+
+@pytest.fixture
+def mock_chirps(mock_country_config):
+    """Create CHIRPS class with mock country config."""
+    geo_bounding_box = GeoBoundingBox(
+        lat_max=6, lat_min=3.2, lon_max=2, lon_min=-3
+    )
+
+    def _mock_chirps(frequency: str = "daily"):
+        if frequency == "daily":
+            chirps = ChirpsDaily(
+                country_config=mock_country_config,
+                geo_bounding_box=geo_bounding_box,
+            )
+        else:
+            chirps = ChirpsMonthly(
+                country_config=mock_country_config,
+                geo_bounding_box=geo_bounding_box,
+            )
+        return chirps
+
+    return _mock_chirps
+
+
+@pytest.fixture
+def mock_download(mocker, mock_chirps):
+    """
+    Call download with mocked _download.
+
+    `forecast_type` is the type of forecast to
+    test, can be either 'prob' or 'dominant'.
+    """
+    download_mock = mocker.patch(
+        "aatoolbox.datasources.chirps.chirps._Chirps._actual_download"
+    )
+
+    def _mock_download(
+        frequency: str,
+        start_year,
+        end_year,
+        start_month,
+        end_month,
+        start_day,
+        end_day,
+    ):
+        chirps = mock_chirps(frequency=frequency)
+        chirps.download(
+            start_year=start_year,
+            end_year=end_year,
+            start_month=start_month,
+            end_month=end_month,
+            start_day=start_day,
+            end_day=end_day,
+        )
+        _, kwargs_download = download_mock.call_args
```

Comment:
Okay sorry, I dont have enough time atm to look at this properly but I think `call_args` returns the last call only? would it instead make more sense to return `call_args_list` as we would generally expect `download_mock` to be called several times?

Reply:
then you could later on test that not only the last date is returned but all that you expect

Reply:
Not sure if this is necessary, as there are already tests on `_create_date_list`. @turnerm what do you think?

Reply:
I think Tinka is correct, her suggestion is really the "right" way to do it. Plus tests should generally be on public methods.